### PR TITLE
test: fix flaky test "Sentry errors after initialization, after opting into metrics should capture UI application state"

### PR DESCRIPTION
**Pull request by chloeYue**
## **Description**

Sentry e2e test "Sentry errors after initialization, after opting into metrics should capture UI application state" starts
to fail because the snapshot is not updated.

Failure link:
https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/85539/workflows/10918e44-d2fe-4384-ab79-d52d4fc85fa8/jobs/3101433/parallel-runs/15

I tried to update the snapshot and saw it's only the order changed that makes the test fails.

Screenshot 2024-06-05 at 17 26 38

So i took advice from PR comment to replace `deepStrictEqual` by `isEqual` in loadsh, which checks the content of the items and permits differences in item order. This way, the test will not fail in the future only because the item order has changed.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25062?quickstart=1)

## **Related issues**

Relates to a previous fix: #24906 

## **Manual testing steps**

Test should pass on all e2e tests.
Run the below command and no errors should appear in codespace or locally
yarn test:e2e:single test/e2e/tests/metrics/errors.spec.js --browser=chrome --debug --leave-running

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by hjetpoluru** [comment]
@chloeYue, small suggestion could we any other assert or below Lodash `const _ = require('lodash');

const json1 = { name: "John", age: 30 };
const json2 = { age: 30, name: "John" };

// Using Lodash's isEqual for deep comparison
if (!_.isEqual(json1, json2)) {
    throw new Error("The JSON objects are not equal.");
}`

**Comment by chloeYue** [comment]
> @chloeYue, a small suggestion: could we use an assertion method that validates the order, or perhaps the following Lodash function could be helpful?
> 
> ```
> const _ = require('lodash');
> 
> const json1 = { name: "John", age: 30 };
> const json2 = { age: 30, name: "John" };
> 
> // Using Lodash's isEqual for deep comparison
> if (!_.isEqual(json1, json2)) {
>     throw new Error("The JSON objects are not equal.");
> }
> ```

Good suggestion, addressed in last commit, please check. @hjetpoluru

**Review by hjetpoluru — DISMISSED** [review]
LGTM! Thanks Chloe.

File: test/e2e/tests/metrics/errors.spec.js [hunk 1]
```diff
@@ -114,7 +114,7 @@ async function matchesSnapshot({
   const snapshotData = JSON.parse(rawSnapshotData);
 
   try {
-    assert.deepStrictEqual(data, snapshotData);
+    isEqual(data, snapshotData);
```
Comment (danjm) [inline test/e2e/tests/metrics/errors.spec.js:117]: If the snapshot data does not match, won't `isEqual` won't throw an error, so this test will pass even if it should fail. We probably need this to be `assert(isEqual(data, snapshotData))`

**Review by danjm — CHANGES_REQUESTED** [review]

**Comment by danjm** [comment]
The failure on circleci seems to be about more than just order of elements. 
```
+ actual - expected ... Lines skipped

  {
    DNS: 'object',
...
      networksMetadata: {
        networkConfigurationId: {
+         EIPS: {},
+         status: 'unknown'
-         EIPS: {
-           '1559': false
-         },
-         status: 'available'
        }
...
      state: 'CLOSED'
      ```
      
      Where were you seeing that the problem is the order of elements?

File: test/e2e/tests/metrics/errors.spec.js (hunk 1 above)
Comment (chloeYue) [reply test/e2e/tests/metrics/errors.spec.js]: Oooooops... sorry, not sure why I removed the assert, it wasn't intentional. I've corrected it in the latest commit. Thanks for the great catch!

**Comment by chloeYue** [comment]
> The failure on circleci seems to be about more than just order of elements.
> 
> ```
> + actual - expected ... Lines skipped
> 
>   {
>     DNS: 'object',
> ...
>       networksMetadata: {
>         networkConfigurationId: {
> +         EIPS: {},
> +         status: 'unknown'
> -         EIPS: {
> -           '1559': false
> -         },
> -         status: 'available'
>         }
> ...
>       state: 'CLOSED'
> ```
> 
> Where were you seeing that the problem is the order of elements?

Seems this isn't the correct link, I couldn't locate the failure related to the order of elements for now. I'll mark this PR as a draft and investigate more recent failures.

**Comment by chloeYue** [comment]
Close this one, as it is addressed in #25137

**Pull request closed without merging**